### DISCUSSION TITLE
Version Engine 1.3.0 is available

### DISCUSSION THREAD

**Discussion by honzasladek** [contember/admin#611] [Announcements]
## 🚀 Major updates in Engine 1.3

After months of development and testing, we are pleased to announce the release of version 1.3. This latest version is packed with exciting new features, performance improvements, bug fixes and much more. Here are just a few of the highlights, you can find the full changelog in the [release notes](https://github.com/contember/engine/releases/tag/v1.3.0).

- Contember Actions: Monitor entity changes and automate workflows via webhooks. This feature is currently in alpha.
[Documentation](https://docs.contember.com/reference/engine/actions/overview)
- connectOrCreate: A single GraphQL mutation allows you to connect to an existing entity or create a new one. Simplifying data management.
- Unified Schema Definition API: The new "c" symbol houses all schema definition APIs, making the schema definition process seamless.
- Improved Orphan Removal: Enhancements ensure that related entities are deleted reliably when the parent entity is removed.
- Granular Invite Permissions: Separate invite permissions from the general "manage" permissions for better control over invitation actions.
- Default "order by" at Entity Level: Specify a default sort order for an entity in the schema, eliminating repetitive sorting definitions during queries.
- Identity Provider (IdP) Updates: Better support for Facebook and Apple ID, plus more flexible parameters in GraphQL methods.
- Rewritten Deletes: Direct database interactions become more predictable with permissions now checked before deletion.

## 🔍 Other Noteworthy Updates

- Tenant API: Various improvements like direct Facebook and Apple ID support, pagination for project members, and improved permissions settings for invites.
- Content API: Revamped delete executor, improved orphan removal, and enforcement of Non-Null constraints on specific mutation inputs.
- CLI: New options added to various commands, and CLI now works in single-project mode.
- Schema, Definition, Migrations: Default order on entity level, better handling of database index and constraint names, and improved schema definition.
- S3: Enhancements for generateUploadUrl mutation and ACL updates to limit file size.
- Enterprise Edition: License updated to allow use for testing purposes.
- Plugins API: New hooks introduced and support for websockets added.

We recommend using version 1.3 for your projects, it's already the default version in Contember Cloud.

**Comment by iBobik** [comment]
Congrats for stable version :-)

I found one surprise for me - Plugins API. What is it? I haven't found docs about it.
